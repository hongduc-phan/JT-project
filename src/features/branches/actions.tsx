import {action} from 'typesafe-actions';

import {
  BRANCH_CREATE,
  BRANCH_DELETE,
  BRANCH_EDIT,
  BRANCH_GET_DETAIL,
  BRANCH_LIST_ALL,
  BRANCH_SET_LIST,
  BRANCH_UPDATE_ENTITY,
} from './constants';
import {FetchCreateBranchBody} from './apis';
import {Branch} from './models';
import {BranchEntity} from './reducers';

export const listBranches = (
  page: number,
  limit: number,
  name: string,
  done?: (success: boolean) => void,
) => action(BRANCH_LIST_ALL, {page, limit, name}, {done});

export interface SetBranchDefaultData {
  data?: Branch[];
  total?: number;
  limit?: number;
  page?: number;
  lastUpdated?: number;
  isFetching?: boolean;
  isFailed?: boolean;
  keyword?: string;
}

export const updateBranchEntity = (id: number, data?: Partial<BranchEntity>) =>
  action(BRANCH_UPDATE_ENTITY, {
    id,
    data,
  });

export const branchSetDefault = (data: SetBranchDefaultData) =>
  action(BRANCH_SET_LIST, data);

export const createBranch = (
  data: FetchCreateBranchBody,
  done?: (success: boolean) => void,
) => action(BRANCH_CREATE, data, {done});

export const deleteBranch = (id: number, done?: (success: boolean) => void) =>
  action(BRANCH_DELETE, id, {done});

export const editBranch = (
  id: number,
  data: FetchCreateBranchBody,
  done?: (success: boolean) => void,
) => action(BRANCH_EDIT, {id, data}, {done});

export const detailBranch = (
  id: number,
  done?: (success: boolean, notFound?: boolean) => void,
) => action(BRANCH_GET_DETAIL, id, {done});
