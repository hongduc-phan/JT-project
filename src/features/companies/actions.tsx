import {action} from 'typesafe-actions';
import {
  COMPANY_ADD_BANK,
  COMPANY_DELETE_BANK,
  COMPANY_EDIT,
  COMPANY_GET_DETAIL_DEFAULT,
  COMPANY_SET_DEFAULT,
} from './constants';
import {CompaniesDefaultState} from './reducers';
import {AddBank} from './models';
import {FetchEditCompanyBody} from './apis';

export const companyDefaultGetDetail = (
  id: string,
  done?: (success: boolean) => void,
) =>
  action(COMPANY_GET_DETAIL_DEFAULT, id, {
    done,
  });

export const companySetDefault = (data: Partial<CompaniesDefaultState>) =>
  action(COMPANY_SET_DEFAULT, data);

export const companyAddBank = (
  data: {
    info: AddBank;
    companyId: string;
  },
  done?: (success: boolean) => void,
) => action(COMPANY_ADD_BANK, data, {done});

export const companyDeleteBank = (
  companyId: string,
  bankAccountId: number,
  done?: (success: boolean) => void,
) => action(COMPANY_DELETE_BANK, {companyId, bankAccountId}, {done});

export const editCompany = (
  companyId: string,
  data: FetchEditCompanyBody,
  done?: (success: boolean) => void,
) => action(COMPANY_EDIT, {companyId, data}, {done});
