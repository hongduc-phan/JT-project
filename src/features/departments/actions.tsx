import {action} from 'typesafe-actions';
import {
  DEPARTMENT_CREATE,
  DEPARTMENT_DELETE,
  DEPARTMENT_EDIT,
  DEPARTMENT_LIST_ALL,
  DEPARTMENT_SET_LIST,
  DEPARTMENT_UPDATE_ENTITY,
} from './constants';
import {DepartmentEntity} from './reducers';
import {Department} from './models';
import {FetchCreateDepartmentBody} from './apis';

export const listDepartments = (
  page: number,
  limit: number,
  name: string,
  done?: (success: boolean) => void,
) => action(DEPARTMENT_LIST_ALL, {page, limit, name}, {done});

export interface SetDepartmentDefaultData {
  data?: Department[];
  total?: number;
  limit?: number;
  page?: number;
  lastUpdated?: number;
  isFetching?: boolean;
  isFailed?: boolean;
  keyword?: string;
}

export const updateBranchEntity = (
  id: number,
  data?: Partial<DepartmentEntity>,
) =>
  action(DEPARTMENT_UPDATE_ENTITY, {
    id,
    data,
  });

export const departmentSetDefault = (data: SetDepartmentDefaultData) =>
  action(DEPARTMENT_SET_LIST, data);

export const createDepartment = (
  data: FetchCreateDepartmentBody,
  done?: (success: boolean) => void,
) => action(DEPARTMENT_CREATE, data, {done});

export const editDepartment = (
  id: number,
  data: Partial<FetchCreateDepartmentBody>,
  done?: (success: boolean) => void,
) => action(DEPARTMENT_EDIT, {id, data}, {done});

export const deleteDepartment = (
  id: number,
  done?: (success: boolean) => void,
) => action(DEPARTMENT_DELETE, {id}, {done});
