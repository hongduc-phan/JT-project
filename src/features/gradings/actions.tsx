import {action} from 'typesafe-actions';

import {
  GRADING_CREATE,
  GRADING_DELETE,
  GRADING_EDIT,
  GRADING_LIST_ALL,
  GRADING_SET_LIST,
  GRADING_UPDATE_ENTITY,
} from './constants';
import {GradingEntity} from './reducers';
import {Grading} from './models';
import {FetchCreateGradingBody} from './apis';

export interface SetGradingDefaultData {
  data?: Grading[];
  total?: number;
  limit?: number;
  page?: number;
  lastUpdated?: number;
  isFetching?: boolean;
  isFailed?: boolean;
  keyword?: string;
}

export const listGradings = (
  page: number,
  limit: number,
  name: string,
  done?: (success: boolean) => void,
) => action(GRADING_LIST_ALL, {page, limit, name}, {done});

export const updateGradingEntity = (
  id: number,
  data?: Partial<GradingEntity>,
) =>
  action(GRADING_UPDATE_ENTITY, {
    id,
    data,
  });

export const gradingSetDefault = (data: SetGradingDefaultData) =>
  action(GRADING_SET_LIST, data);

export const createGrading = (
  data: FetchCreateGradingBody,
  done?: (success: boolean) => void,
) => action(GRADING_CREATE, data, {done});

export const editGrading = (
  id: number,
  data: Partial<FetchCreateGradingBody>,
  done?: (success: boolean) => void,
) => action(GRADING_EDIT, {id, data}, {done});

export const deleteGrading = (id: number, done?: (success: boolean) => void) =>
  action(GRADING_DELETE, {id}, {done});
