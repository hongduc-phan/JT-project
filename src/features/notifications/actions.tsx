import {action} from 'typesafe-actions';
import {ReactNode} from 'react';
import {SNACK_ADD, SNACK_REMOVE} from './constants';
import {SnackbarVariant} from '../../components/Snackbar';

export const snackAdd = (msg: ReactNode, variant?: SnackbarVariant) =>
  action(SNACK_ADD, {msg, variant});

export const snackRemove = (key: number) => action(SNACK_REMOVE, key);
