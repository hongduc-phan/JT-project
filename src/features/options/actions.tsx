import {action} from 'typesafe-actions';
import {
  BANKS_GET,
  BANKS_SET,
  COUNTRIES_GET,
  COUNTRIES_SET,
  STATES_GET,
  STATES_SET,
} from './constants';
import {BanksState, CountriesState, CountryStatesState} from './reducers';

export const countriesGet = () => action(COUNTRIES_GET);
export const countriesSet = (data: Partial<CountriesState>) =>
  action(COUNTRIES_SET, data);

export const statesGet = (countryId?: string) => action(STATES_GET, countryId);
export const statesSet = (data: Partial<CountryStatesState>) =>
  action(STATES_SET, data);

export const banksGet = (stateId?: number) => action(BANKS_GET, stateId);
export const banksSet = (data: Partial<BanksState>, stateId?: number) =>
  action(BANKS_SET, {
    data,
    stateId,
  });
