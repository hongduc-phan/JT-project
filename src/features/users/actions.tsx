import {action} from 'typesafe-actions';
import {LOGIN, LOGOUT, SET} from './constants';
import {UserInformation} from './models';

export const login = (
  company: string,
  username: string,
  password: string,
  done: (reset?: boolean) => void,
) =>
  action(
    LOGIN,
    {company, username, password},
    {
      done,
    },
  );

export const logout = () => action(LOGOUT);

export const set = (data: UserInformation) => action(SET, data);
