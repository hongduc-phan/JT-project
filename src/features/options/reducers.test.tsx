import {
  initialOptionsBanksState,
  initialOptionsCountriesState,
  initialOptionsCountryStatesState,
  OptionsStates,
} from './reducers';
import {optionsReducers} from './index';
import {banksSet, countriesSet} from './actions';

const getInitState = (state?: Partial<OptionsStates>) => {
  return optionsReducers(
    {
      countries: initialOptionsCountriesState,
      states: initialOptionsCountryStatesState,
      banks: initialOptionsBanksState,
      ...state,
    },
    {} as any,
  );
};

describe('Options Reducers', () => {
  describe('Init state', () => {
    it('Should match snapshot', () => {
      expect(getInitState()).toMatchSnapshot();
    });
  });

  describe('Countries', () => {
    it('Should update new state', () => {
      const initState = getInitState();
      expect(initState.countries).toEqual(initialOptionsCountriesState);
      const state = optionsReducers(
        initState,
        countriesSet({
          isFetching: true,
          isFailed: true,
          list: [{countryCode: '500', countryId: 500, countryName: 'SG'}],
        }),
      );
      expect(state.countries.isFailed).toBeTruthy();
      expect(state.countries.isFetching).toBeTruthy();
      expect(state.countries.list).toHaveLength(1);
      expect(state.countries.lastUpdated).not.toEqual(
        initState.countries.lastUpdated,
      );
    });
  });

  describe('banks', () => {
    it('Should set new state when list empty', () => {
      const initState = getInitState();
      expect(initState.banks).toEqual(initialOptionsBanksState);
      const state = optionsReducers(
        initState,
        banksSet({
          isFailed: true,
          isFetching: true,
          list: [{stateId: 1, bankName: '1', bankId: 1, lastUpdated: 0}],
        }),
      );

      expect(state.banks.isFailed).toBeTruthy();
      expect(state.banks.isFetching).toBeTruthy();
      expect(state.banks.list).toHaveLength(1);
      expect(state.banks.lastUpdated).not.toEqual(initState.banks.lastUpdated);
    });

    it('Should update banks information', () => {
      const initState = getInitState({
        banks: {
          isFetching: false,
          isFailed: false,
          list: [{stateId: 1, bankName: '1', bankId: 1, lastUpdated: 0}],
          lastUpdated: 0,
        },
      });
      expect(initState.banks.list[0].bankName).toEqual('1');
      const state = optionsReducers(
        initState,
        banksSet(
          {
            isFailed: true,
            isFetching: true,
            list: [{stateId: 1, bankName: '2', bankId: 1, lastUpdated: 0}],
          },
          1,
        ),
      );

      expect(state.banks.list[0].bankName).toEqual('2');
    });
  });
});
