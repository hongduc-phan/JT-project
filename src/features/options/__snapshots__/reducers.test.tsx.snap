// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Reducers Init state Should match snapshot 1`] = `
Object {
  "banks": Object {
    "isFailed": false,
    "isFetching": false,
    "lastUpdated": 0,
    "list": Array [],
  },
  "countries": Object {
    "isFailed": false,
    "isFetching": false,
    "lastUpdated": 0,
    "list": Array [],
  },
  "states": Object {
    "isFailed": false,
    "isFetching": false,
    "lastUpdated": 0,
    "list": Array [],
  },
}
`;
